### PR TITLE
Update createjobform to typescript

### PR DESCRIPTION
**Pull request by Pespiri**

### CODE REVIEW

**Review by satr — DISMISSED** [review]
Minor questions.

File: src/components/job-overview/steps-list.js [hunk 1]
```diff
@@ -7,29 +7,32 @@ import {
...
-    step.name === PipelineStep.CloneConfig ||
-    step.name === PipelineStep.CloneRepository
+    name === PipelineStep.CloneConfig ||
+    name === PipelineStep.CloneRepository
   ) {
     return github;
-  } else if (step.name === PipelineStep.CloneConfigToMap) {
-    return copy; //outdated, needed for old jobs
-  } else if (step.name === PipelineStep.PreparePipelines) {
+  } else if (
```
[the remark this answers is not in the document]
Comment (Pespiri) [reply src/components/job-overview/steps-list.js:23]: This is not an `else`, it's an `else if`

File: src/state/job-creation/action-creators.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+export const actions = {
+  /**
+   * Action creator for requesting an job creation
+   * @param {Object} job The job object
+   */
+  addJobRequest: makeActionCreator<
+    never,
+    { job: Parameters<typeof createJob> },
+    typeof actionTypes.JOB_CREATION_REQUEST
+  >(actionTypes.JOB_CREATION_REQUEST, 'job'),
```
[the remark this answers is not in the document]
Comment (Pespiri) [reply src/state/job-creation/action-creators.ts]: I've been working on some typing, but do not intend to commit it just yet. Just didn't notice I filled it in on this component. Will clear it up

**Review by satr — APPROVED** [review]

File: src/components/create-job-form/pipeline-form-promote.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        // of the newly-selected deployment then we invalidate the form
+        const selectedEnv = environments.find((x) => x.name === toEnvironment);
+        isValid = !!(selectedEnv?.activeDeployment?.name === value
+          ? false
+          : toEnvironment && value);
+
+        // When selecting a deployment to promote we need to add its environment
+        // to the state that is sent to the API (the "fromEnvironment" argument)
+        const selectedDeployment = deployments.find((x) => x.name === value);
+        newState.fromEnvironment = selectedDeployment?.environment;
```
[the remark this answers is not in the document]
Comment (Pespiri) [reply src/components/create-job-form/pipeline-form-promote.tsx:48]: fixed

**Review by satr — APPROVED** [review]

**Pull request merged**